obd_ontology/extract_dtcs_and_components_from_excel_file.py

```diff
@@ -0,0 +1,131 @@
+import argparse
+import pandas
+import numpy as np
+
+from expert_knowledge_enhancer import ExpertKnowledgeEnhancer
+
+valid_special_characters = " ,()-:&/"
+
+expert_knowledge_enhancer = ExpertKnowledgeEnhancer()
+
+
+def create_dtc_dictionary(path: str) -> dict:
+    """
+    Reads the MSI excel file and extracts the relevant information concerning DTCs and components.
+
+    :param path: path to the excel file
+    :return: dictionary with DTCs as keys and a list containing fault condition and tuples of position and
+    component as values, e.g. [fault_cond, [1, comp1], [2, comp2]]
+    """
+    dtc_dict = {}
+
+    data = pandas.read_excel(path, sheet_name="DTC - Element ID - Baum")
+
+    for i in range(len(data)):
+
+        dtc = str(data["DTC"][i])
+        pos = data["Ursachenposition"][i]
+        comp = str(data["Element ID"][i])
+        fault_cond = str(data["Fehlercodes"][i])
+        alternative_fault_cond = str(data["Alternative Klartexte der P0 Codes"][i])
+        klavkarr_fault_cond = str(data["klavkarr Fehlercodes"][i])
+        british_fault_cond = str(data["British Standard ISO 15031-6 (2005)"][i])
+
+        if dtc != "nan":
+            if dtc not in dtc_dict:
+                if fault_cond != "nan":
+                    dtc_dict[dtc] = [fault_cond]
+                elif alternative_fault_cond != "nan":
+                    dtc_dict[dtc] = [alternative_fault_cond]
+                elif klavkarr_fault_cond != "nan":
+                    dtc_dict[dtc] = [klavkarr_fault_cond]
+                elif british_fault_cond != "nan":
+                    dtc_dict[dtc] = [british_fault_cond]
+                else:
+                    dtc_dict[dtc] = [""]
+
+            if comp != "nan":
+                dtc_dict[dtc].append([pos, comp])
```

Comment:
First, you are not creating tuples here, but lists. Second, why do you need the `pos` at all, isn't this always just the index of the component in that list?